src/library/masq.js

```diff
@@ -414,30 +440,62 @@ class Masq {
         try {
           if (app) {
             const privateProfile = await this.getProfile(this.profileId)
-            await sendAuthorized(peer, app.id, app.appDEK, privateProfile.username, privateProfile.image, app.appNonce)
+            const res = await this.sendAuthorized(peer, app.id, app.appDEK, privateProfile.username, privateProfile.image, app.appNonce)
+
+            resolve(res)
           } else {
-            await sendNotAuthorized(peer)
+            const res = await this.sendNotAuthorized(peer)
+            resolve(res)
           }
         } catch (e) {
           await this._closeUserAppConnection()
-          return reject(new MasqError(MasqError.DISCONNECTED_DURING_LOGIN))
+          return reject(e)
         }
+      })
+    })
+  }
 
-        peer.once('data', async (data) => {
-          const json = await decrypt(this.key, JSON.parse(data), 'base64')
+  async sendAuthorized (peer, userAppDbId, userAppDEK, username, profileImage, userAppNonce) {
+    return new Promise(async (resolve, reject) => {
+      this.setState(STATES.LOGGED)
+      const data = { msg: 'authorized', userAppDbId, userAppDEK, username, profileImage, userAppNonce }
+      const encryptedMsg = await encrypt(this.key, data, 'base64')
+      peer.send(JSON.stringify(encryptedMsg))
+      peer.once('data', async (data) => {
+        const json = await decrypt(this.key, JSON.parse(data), 'base64')
 
-          if (json.msg === 'connectionEstablished') {
-            await this._closeUserAppConnection()
-            return resolve({ isConnected: true })
-          } else if (json.msg === 'registerUserApp') {
+        if (json.msg === 'connectionEstablished') {
+          await this._closeUserAppConnection()
+          resolve({ isConnected: true })
+        } else {
+          await this._closeUserAppConnection()
+          reject(new MasqError(MasqError.WRONG_MESSAGE, `Unexpectedly received message with type ${json.msg}`))
+        }
+      })
+    })
+  }
+
+  async sendNotAuthorized (peer) {
```

Comment:
same thing here, the sole purpose of `sendNotAuthorized` could be to send the message

Reply:
ok